### ISSUE TITLE
Upgrade to jinja 3.x

### ISSUE THREAD

**Issue by frankplow** [qutebrowser/qutebrowser#7090]
qutebrowser is currently using jinja2, which has been deprecated for some time now. [This has culminated in a recent update to markupsafe breaking jinja2](https://github.com/pallets/jinja/issues/1585). In turn, the Arch (and presumably other rolling release) package is currently broken.

**Comment by The-Compiler** [comment]
The package is still called Jinja2. qutebrowser upgrades its dependencies weekly, and has switched to Jinja2 3.0 less than a week after it was released, almost a year ago.

**Comment by frankplow** [comment]
My apologies, jinja2 v3.x is a bit of a confusing title. My point about the Arch package being broken at present still stands, though it looks as if this is an issue with arch's python-jinja package rather than qutebrowser.

**Comment by The-Compiler** [comment]
What does "it's broken" mean precisely? qutebrowser still runs fine for me after upgrading, so not much I can say without more information.

**Comment by frankplow** [comment]
Attempting to launch qutebrowser gives the
`ImportError: cannot import name 'soft_unicode' from 'markupsafe' (/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/markupsafe/__init__.py)
Error: Process completed with exit code 1.`
from the linked issue above, which stems from using a deprecated version of jinja. I think that Arch's python-jinja package is somehow incorrectly configured, as updating it didn't seem to update the version of Jinja2 shown in pip. I've managed to fix the issue by uninstalling and then reinstalling python-jinja.

**Comment by The-Compiler** [comment]
Archlinux also [upgraded its package to jinja2 v3](https://github.com/archlinux/svntogit-community/commit/a156bcc0eaa099041db5cbb8ee6cc8fd111f3f77) back in May 2021. I can't say more without seeing the exact path (`/opt/hostedtoolcache/` is on GitHub Actions, not Archlinux), but chances are that you had an old copy of Jinja2 on your system which isn't coming from the Archlinux repositories (e.g. because you used `sudo pip install` or even `pip install --user` at some point).

**Comment by LyndonArmitage** [comment]
Ran into this issue today after an upgrade. Reinstalling the latest versions of either markupsafe or jinja does not solve the issue for me.

The error message states the problem: `ImportError: cannot import name 'soft_unicode' from 'markupsafe'`

Looking at the [change log for markupsafe](https://markupsafe.palletsprojects.com/en/2.1.x/changes/#version-2-1-0) 

> Remove soft_unicode, which was previously deprecated. Use soft_str instead. [#261](https://github.com/pallets/markupsafe/pull/261)

A quick search on Gituhub shows this is not used directly by Qutebrowser so it must be a library issue. I did a search of my installed jinja2 package and it does not mention `soft_unicode` so I am a bit stumped. I am using the community package in Manjaro (should be the Arch community version 2.5.0-1).
